sdk/tables/azure-data-tables/azure/data/tables/_serialize.py

```diff
@@ -226,7 +226,7 @@ def _add_entity_properties(source):
         elif isinstance(value, datetime):
             mtype, value = _to_entity_datetime(value)
         elif isinstance(value, EntityProperty):
-            conv = _EDM_TO_ENTITY_CONVERSIONS.get(value.type)
+            conv = _EDM_TO_ENTITY_CONVERSIONS.get(value.edm_type)
```

Comment:
We should remove the `isinstance` check, and instead treat this as a generic tuple....
```python
elif isinstance(value, tuple):  # ideally we wouldn't do the instance check at all here.... 
    conv = _EDM_TO_ENTITY_CONVERSIONS.get(value[1])
    ...
    mtype, value = conv(value[0])
```